dvc/repo/add.py

```diff
@@ -3,52 +3,75 @@
 import os
 import logging
 import colorama
+from six import string_types
 
 from dvc.repo.scm_context import scm_context
 from dvc.stage import Stage
 from dvc.utils import walk_files, LARGE_DIR_SIZE
 from dvc.exceptions import RecursiveAddingWhileUsingFilename
-
+from dvc.progress import Tqdm
 from . import locked
 
-
 logger = logging.getLogger(__name__)
 
 
 @locked
 @scm_context
-def add(repo, target, recursive=False, no_commit=False, fname=None):
+def add(
+    repo,
+    target_list,
+    recursive=False,
+    no_commit=False,
+    fname=None,
+    progress=False,
+):
     if recursive and fname:
         raise RecursiveAddingWhileUsingFilename()
 
-    targets = _find_all_targets(repo, target, recursive)
+    if isinstance(target_list, string_types):
+        target_list = [target_list]
 
-    if os.path.isdir(target) and len(targets) > LARGE_DIR_SIZE:
-        logger.warning(
-            "You are adding a large directory '{target}' recursively,"
-            " consider tracking it as a whole instead.\n"
-            "{purple}HINT:{nc} Remove the generated DVC-file and then"
-            " run {cyan}dvc add {target}{nc}".format(
-                purple=colorama.Fore.MAGENTA,
-                cyan=colorama.Fore.CYAN,
-                nc=colorama.Style.RESET_ALL,
-                target=target,
-            )
-        )
+    stages_list = []
+    with Tqdm(
+        total=len(target_list),
+        desc="Adding",
```

Comment:
It is weird that this is only shown for 0%, after which it simply shows file names.

Reply:
yes but after the bar is completed it goes back to showing the initial message (i.e. "Adding").
I was thinking of changing it to just "Add" from the start though.